bin/v2.js

```diff
@@ -525,8 +555,111 @@ const releaseCmd = (cmdObj, bucketNames, lambdaFunctions, environments, bucketPa
+
+        requestCurrentlyReleasedVersions(environments).then(releasedVersions => {
+          return getDeployedVersionListWithMetadata(bucketName, bucketPath, packageName).then(deployedArtifacts => {
+
+            console.log(`There are currently a total of ${deployedArtifacts.length} deployed artifacts in ${serverEnvironment}.`);
+
+            const oldArtifacts = deployedArtifacts
+              .filter(artifact => {
+                // only allow an asset to be deleted if it's older than a year
+                return artifact.lastModified.getTime() < ninetyDaysAgo.getTime()
+              })
+              .filter(artifact => {
+                // only allow an asset to be deleted if it's not being used in any of the environments
+                for (const releasedVersion of releasedVersions) {
+                  // noinspection JSUnresolvedVariable
+                  if (releasedVersion.releasedHash === artifact.hash) {
+                    return false;
+                  }
+                }
+                return true;
+              })
+            console.log(`We are going to delete ${oldArtifacts.length} ${packageName} artifacts in ${serverEnvironment}`);
+
+            return oldArtifacts;
+          })
+        }).then(artifactsToDelete => {
+
+          if (artifactsToDelete.length < 1) {
+            console.log("Nothing to delete.")
+          }
+
+          for (const artifact of artifactsToDelete) {
+            console.log(artifact.displayName);
+          }
+          inquirer.prompt({
+            type: 'confirm',
+            name: "reallyDelete",
+            message: `Really delete these assets in the S3 bucket ${bucketName} with the path: ${bucketPath}?`,
+            default: false
+          }).then(answers => {
+            // noinspection JSUnresolvedVariable
+            if (answers.reallyDelete) {
+              for (const artifact of artifactsToDelete) {
+
+                console.log("Deleting", artifact.hash);
+
+                const s3 = new AWS.S3({
+                  apiVersion: '2006-03-01'
+                });
+
+                emptyS3Directory(s3, bucketName, artifact.directoryPath);
+
+                const paramsOfObjectsToDelete = {
+                  Bucket: bucketName, Delete: {
+                    Objects: [{
+                      Key: artifact.filepath
+                    }, {
+                      Key: artifact.directoryPath
+                    }],
+                  }
+                };
+
+                s3.deleteObjects(paramsOfObjectsToDelete, function (err, data) {
+                  if (err) console.error(err, err.stack); // an error occurred
+                  else console.log(data);           // successful response
+                });
+              }
+              console.log("Deleted.")
+            } else {
+              console.log("Very well, not deleting anything then.")
+            }
+          })
+
```

Comment:
I'd prefer no newline, but I don't do much JS at all, so take that with a grain of salt
```suggestion
```